### ISSUE TITLE
TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by evilebottnawi** [tclindner/npm-package-json-lint#53]
* A little info about your environment
    * npm-package-json-lint version `2.8.0`
    * npm version `5.3.0`
    * node version `8.4.0`
  * The output from npm-package-json-lint
```shell
TypeError: Cannot read property 'length' of undefined
/path/to/node_modules/npm-package-json-lint/src/cli.js:28
  throw new Error(err);
  ^

Error: TypeError: Cannot read property 'length' of undefined
    at handleError (/path/to/node_modules/npm-package-json-lint/src/cli.js:28:9)
    at Object.<anonymous> (/path/to/node_modules/npm-package-json-lint/src/cli.js:126:3)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Function.Module.runMain (module.js:609:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:598:3

```
  * What you expected to happen
Not throw error.
  * The steps to reproduce the problem
1. Create `npmpackagejsonlint.config.js` with `prefer-property-order` rule.
```shell
module.exports = {
  rules: {
    "prefer-property-order": ["error", [
      "name",
      "version",
      "description",
      "keywords",
      "homepage",
      "bugs",
      "license",
      "author",
      "contributors",
      "files",
      "main",
      "jsnext:main",
      "bin",
      "man",
      "directories",
      "repository",
      "scripts",
      "config",
      "dependencies",
      "devDependencies",
      "peerDependencies",
      "bundleDependencies",
      "bundledDependencies",
      "optionalDependencies",
      "engines",
      "engineStrict",
      "os",
      "cpu",
      "preferGlobal",
      "private",
      "publishConfig"
    ]]
  }
};
```
2. Create `package.json`.
```json
{
  "name": "test",
  "version": "0.0.4",
  "description": "test"
}
```

**Comment by tclindner** [comment]
Thank you for catching this, @evilebottnawi! I've release v2.8.1. It should address your issue. It also contains feedback from #50. Let me know what you think.

**Comment by evilebottnawi** [comment]
@tclindner problem still exists :disappointed: use post above to reproducible error.
In `function(packageJsonData, lintType, preferredOrder)` `preferredOrder` contains `undefined`, but `lintType` containts
```shell
[ 'error',
  [ 'name',
    'version',
    'description',
    'keywords',
    'homepage',
    'bugs',
    'license',
    'author',
    'contributors',
    'files',
    'main',
    'jsnext:main',
    'bin',
    'man',
    'directories',
    'repository',
    'scripts',
    'config',
    'dependencies',
    'devDependencies',
    'peerDependencies',
    'bundleDependencies',
    'bundledDependencies',
    'optionalDependencies',
    'engines',
    'engineStrict',
    'os',
    'cpu',
    'preferGlobal',
    'private',
    'publishConfig' ] ]
```

**Comment by tclindner** [comment]
I'm so sorry about that @evilebottnawi! I believe I have identified the issue. The rules loader was not properly interpreting this rule as a "array" type rule. I've released v2.8.2 to address that. Can you please try it out and see if it resolves your issue? Sorry again about the issues with this rule. 

P.S. I will be logging an enhancement to improve the rules loader to avoid hitting this issue in the future. I'll tag you on that ticket. Thank you again for helping improve this project 🥇
